cmd/worker/internal/ratelimit/handler.go

```diff
@@ -2,103 +2,49 @@ package ratelimit
 
 import (
 	"context"
-	"math"
 	"time"
 
 	"github.com/sourcegraph/log"
+
 	"github.com/sourcegraph/sourcegraph/internal/conf"
 	"github.com/sourcegraph/sourcegraph/internal/database"
-	"github.com/sourcegraph/sourcegraph/internal/extsvc"
 	"github.com/sourcegraph/sourcegraph/internal/goroutine"
 	"github.com/sourcegraph/sourcegraph/internal/ratelimit"
-	"github.com/sourcegraph/sourcegraph/internal/types"
-	"github.com/sourcegraph/sourcegraph/lib/errors"
-	"golang.org/x/time/rate"
-)
-
-var (
-	defaultAPIReplenishmentInterval = int32(3600)
-	defaultGitReplenishmentInterval = int32(1)
 )
 
 var _ goroutine.Handler = &handler{}
 
 type handler struct {
-	codeHostStore database.CodeHostStore
-	rateLimiter   ratelimit.CodeHostRateLimiter
-	logger        log.Logger
+	externalServiceStore database.ExternalServiceStore
+	logger               log.Logger
+	newRateLimiterFunc   func(bucketName string) ratelimit.GlobalLimiter
 }
 
-func (h *handler) Handle(ctx context.Context) error {
-	codeHosts, _, err := h.codeHostStore.List(ctx, database.ListCodeHostsOpts{})
-	if err != nil {
-		return err
-	}
+func (h *handler) Handle(ctx context.Context) (err error) {
+	defer func() {
+		// Be very vocal about these issues.
+		if err != nil {
+			h.logger.Error("failed to sync rate limit configs to redis", log.Error(err))
+		}
+	}()
 
-	// TODO: @varsanojidan This is only needed before the OOB migration, once the OOB migration is done, we can remove this
 	var defaultGitQuota int32
```

Comment:
Is this even a `defaultGitQuota` ? Isn't it just a `gitQuota`

```suggestion
	var defaultGitQuota int32 = 1
```

Can also do that to remove the else branch